README.md
# Persons
Manage persons. This is an extension for the TYPO3 CMS.

### Table of contents

* Models
    * [Content](./Documentation/Models/Content.md)
    * [Person](./Documentation/Models/Content.md)
* Plugins
    * Person
### Credentials
Icons made by <a href="http://www.freepik.com" title="Freepik">Freepik</a> from <a href="http://www.flaticon.com" title="Flaticon">www.flaticon.com</a> is licensed by <a href="http://creativecommons.org/licenses/by/3.0/" title="Creative Commons BY 3.0" target="_blank">CC 3.0 BY</a></div>


ChangeLog
2017-09-29         [RELEASE] Release of persons 0.5.0 (Dirk Wenzel)
2017-09-29 62c00f5 [TASK] cast to bool replaced by !empty() in PersonRepository->findMultipleByUid (Dirk Wenzel)
2017-09-26 83b4352 [BUGFIX] Allow persons with default language in showSelectedAction (Nicole Cordes)

2017-09-26         [RELEASE] Release of persons 0.4.1 (Dirk Wenzel)
2017-09-26 46cf6d6 [BUGFIX] fixed error in AbstractJsonView caused by invalid return value null of TYPO3\CMS\Extbase\Persistence\Generic\LazyLoadingProxy->_loadRealInstance() (Dirk Wenzel)

2017-09-25         [RELEASE] Release of persons 0.4.0 (Dirk Wenzel)
2017-09-20 a2d564d [TASK] german localization for backend added. Changed language file format to xliff. (Dirk Wenzel)

2017-09-19         [RELEASE] Release of persons 0.3.0 (Dirk Wenzel)
2017-09-19 d8a8945 [TASK] german localization for backend added. (Dirk Wenzel)

2017-09-19         [RELEASE] Release of persons 0.2.0 (Dirk Wenzel)
2017-09-14 f0d1178 [BUGFIX] fixed error in configuration of json list view which prevented image field from rendering (Dirk Wenzel)
2017-09-14 6a6e2a6 [BUGFIX] View ListJson now renders missing field uid too. (Dirk Wenzel)
2017-09-13 fd34dd2 [TASK] PersonController->listAction now emits a signal before assigning variables to view too. (Dirk Wenzel)
2017-09-08 8d05a6a [TASK] Json view for person listing extended: now category parent is displayed too. (Dirk Wenzel)
2017-09-08 1218770 [TASK] PersonRepositoryTrait and SignalTrait implemented. PersonController refactored and extended: filterAction now sends signal before assigning variables to view. (Dirk Wenzel)
2017-08-23 cfa2cb0 [TASK] Json View now transforms object storages too. View ListJson renders categories. (Dirk Wenzel)
2017-08-03 03c8f8b [TASK] model Person edited: field status removed, departments replaced by categories (Dirk Wenzel)
2017-08-01 c8d4301 [BUGFIX] fix error in Person model where getter return wrong property. Failing unit test for PersonController fixed. (Dirk Wenzel)
2017-07-31 d6e76e7 [TASK] plugin now displays field pluginNamespace for each action (Dirk Wenzel)
2017-07-31 4d96828 [TASK] show pages configuration for all actions of person plugin (Dirk Wenzel)
2017-07-31 330337f [TASK] Variable assignment of filter action in PersonController changed. Json view configuration adapted. (Dirk Wenzel)
2017-07-31 46944e0 [TASK] Json view now transforms ObjectStorage objects to array (Dirk Wenzel)
2017-07-26 1ef7388 [TASK] Filter action for PersonController implemented (Dirk Wenzel)
2017-07-05 3b01902 [TASK] AbstractJsonView, Person/ListJson and Person/ShowSelectedJson of PersonController implemented. (Dirk Wenzel)
2017-07-04 322b083 [TASK] JsonView implemented (WIP) (Dirk Wenzel)
2017-07-03 49fa06e [TASK] missing property www added to Person. (Dirk Wenzel)

2017-06-29         [RELEASE] Release of persons 0.1.0 (Dirk Wenzel)
2017-06-29 631ee5c [BUGFIX] fixed error in composer.json which caused failing builds (Dirk Wenzel)
2017-06-22 7cccd20 [BUGFIX] invalid annotation in domain model Person fixed. (Dirk Wenzel)
2017-06-22 9ec26f1 [TASK] domain model Person extended by field departments. (Dirk Wenzel)
2017-06-21 22ec2cd [BUGFIX] TypoScript path corrected (Dirk Wenzel)
2017-06-16 6c44e6f [TASK] PersonRepository extended by methods findMultipleByUid and createOrderingsFromList. PersonController extended by showSelectedAction. Plugin can now display selected persons. (Dirk Wenzel)
2017-06-01 1cc6f6e [TASK] field www added to person model (Dirk Wenzel)
2017-05-31 eb5e81c [TASK] icons for extension, plugin and model added. (Dirk Wenzel)
2017-05-29 8fafe15 [BUGFIX] missing label added to table configuration (Dirk Wenzel)
2017-05-29 17042a0 [TASK] fixed error caused by invalid SQL statement. (Dirk Wenzel)
2017-05-29 11ec46a [BUGFIX] invalid configuration fixed: field department replaced by field status (Dirk Wenzel)
2017-05-22 025176f [TASK] Documentation added. SQL definition improved. (Dirk Wenzel)
2017-05-17 9a5fd23 [TASK] Plugin configured, default templates edited. (Dirk Wenzel)
2017-05-17 5e34127 [TASK] Domain/Model/Content implemented. Person extended by field contentElements. Code cleanup. (Dirk Wenzel)
2017-05-16 0052460 [TASK] Basic extension configuration added. Person, PersonController, EmConfiguration, EmConfigurationUtility implemented. (Dirk Wenzel)


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "persons".
 *
 * Auto generated 26-09-2017 17:19
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'Persons',
  'description' => 'Manage persons',
  'category' => 'plugin',
  'author' => 'Dirk Wenzel',
  'author_email' => 'wenzel@cps-it.de',
  'state' => 'beta',
  'uploadfolder' => '1',
  'createDirs' => '',
  'clearCacheOnLoad' => 0,
  'version' => '0.5.0',
  'constraints' =>
  array (
    'depends' =>
    array (
      'typo3' => '8.7.0-8.7.99',
    ),
    'conflicts' =>
    array (
    ),
    'suggests' =>
    array (
    ),
  ),
  '_md5_values_when_last_written' => 'a:54:{s:9:"ChangeLog";s:4:"4934";s:9:"README.md";s:4:"2dab";s:13:"composer.json";s:4:"f35b";s:12:"ext_icon.png";s:4:"7259";s:17:"ext_localconf.php";s:4:"7045";s:14:"ext_tables.php";s:4:"0f33";s:14:"ext_tables.sql";s:4:"39b7";s:24:"ext_typoscript_setup.txt";s:4:"dbfd";s:39:"Classes/Controller/PersonController.php";s:4:"fcef";s:44:"Classes/Controller/PersonRepositoryTrait.php";s:4:"e3a4";s:34:"Classes/Controller/SignalTrait.php";s:4:"b020";s:32:"Classes/Domain/Model/Content.php";s:4:"7075";s:31:"Classes/Domain/Model/Person.php";s:4:"eb4b";s:44:"Classes/Domain/Model/Dto/EmConfiguration.php";s:4:"530d";s:46:"Classes/Domain/Repository/PersonRepository.php";s:4:"6402";s:42:"Classes/Utility/EmConfigurationUtility.php";s:4:"7380";s:33:"Classes/View/AbstractJsonView.php";s:4:"480f";s:34:"Classes/View/Person/FilterJson.php";s:4:"fa8d";s:32:"Classes/View/Person/ListJson.php";s:4:"cf1a";s:32:"Classes/View/Person/ShowJson.php";s:4:"15be";s:40:"Classes/View/Person/ShowSelectedJson.php";s:4:"2f2c";s:44:"Configuration/FlexForms/flexform_persons.xml";s:4:"8375";s:52:"Configuration/TCA/tx_persons_domain_model_person.php";s:4:"8c42";s:42:"Configuration/TCA/Overrides/tt_content.php";s:4:"9bbd";s:62:"Configuration/TCA/Overrides/tx_persons_domain_model_person.php";s:4:"0e50";s:47:"Configuration/TSconfig/ContentElementWizard.txt";s:4:"8b08";s:37:"Configuration/TypoScript/constants.ts";s:4:"3a6c";s:33:"Configuration/TypoScript/setup.ts";s:4:"621e";s:31:"Documentation/Models/Content.md";s:4:"f705";s:30:"Documentation/Models/Person.md";s:4:"21cd";s:32:"Documentation/Plugins/Persons.md";s:4:"3f3d";s:46:"Resources/Private/Language/de.locallang_be.xlf";s:4:"b81c";s:46:"Resources/Private/Language/de.locallang_db.xlf";s:4:"83b2";s:40:"Resources/Private/Language/locallang.xlf";s:4:"443a";s:43:"Resources/Private/Language/locallang_be.xlf";s:4:"03fa";s:78:"Resources/Private/Language/locallang_csh_tx_hfmpersons_domain_model_person.xlf";s:4:"41b8";s:43:"Resources/Private/Language/locallang_db.xlf";s:4:"7334";s:38:"Resources/Private/Layouts/Default.html";s:4:"20c0";s:47:"Resources/Private/Partials/Person/ListItem.html";s:4:"16ff";s:49:"Resources/Private/Partials/Person/Properties.html";s:4:"bafb";s:46:"Resources/Private/Templates/Person/Filter.html";s:4:"c182";s:44:"Resources/Private/Templates/Person/List.html";s:4:"8b5f";s:44:"Resources/Private/Templates/Person/Show.html";s:4:"a8c0";s:52:"Resources/Private/Templates/Person/ShowSelected.html";s:4:"f94d";s:39:"Resources/Public/Icons/icon-persons.png";s:4:"2444";s:39:"Resources/Public/Icons/icon-persons.svg";s:4:"0c36";s:57:"Resources/Public/Icons/tx_persons_domain_model_person.svg";s:4:"3d45";s:46:"Tests/Unit/Controller/PersonControllerTest.php";s:4:"4d1e";s:51:"Tests/Unit/Controller/PersonRepositoryTraitTest.php";s:4:"0888";s:39:"Tests/Unit/Domain/Model/ContentTest.php";s:4:"c796";s:38:"Tests/Unit/Domain/Model/PersonTest.php";s:4:"57ac";s:51:"Tests/Unit/Domain/Model/Dto/EmConfigurationTest.php";s:4:"9a1e";s:53:"Tests/Unit/Domain/Repository/PersonRepositoryTest.php";s:4:"ed4e";s:40:"Tests/Unit/View/AbstractJsonViewTest.php";s:4:"8745";}',
);

